### ISSUE TITLE
Tally triggers fail with more than one filter

### ISSUE THREAD

**Issue by wbinventor** [mit-crpg/openmc#480]
As part of the conversation on PR #471, I've come to the realization that tally triggers do not work with more than one filter. In the case of the MGXS module, tallies have a filter over "cell" / "material" / "universe" IDs and another over "energy" (and in some cases a third over "energyout"). None of these triggers are checked properly since it appears that the logic to compute filter bins fails. In particular, I think there is an issue with lines 174-184 in "src/trigger.F90" ([link](https://github.com/mit-crpg/openmc/blob/develop/src/trigger.F90#L178)). I've always found the code to compute filter indices tricky, and don't have much time at the moment to spend on this, so wanted to post it here to keep a record of it and why triggers won't yet work for the MGXS module.

**Comment by wbinventor** [comment]
I think I've found a pretty easy solution to the multiple filters problem, and one which greatly simplifies the code to check triggers. Unfortunately, I've exposed a new bug in which tally moments (e.g., score "scatter-P1") are not treated properly any more. I think this is due to some changes that @paulromano may have made in which the moment scores are explicitly expanded into multiple score strings, but I'm not sure.
